src/utilities/clipboard-utils.ts

```diff
@@ -1,21 +1,47 @@
 import { gImageMap, ImageMapEntryType } from "../models/image-map";
 
 interface IOnCompleteParams {
-  file: File;
   image: ImageMapEntryType;
 }
 type OnComplete = (params: IOnCompleteParams) => void;
 
-export const pasteClipboardImage = async (onComplete: OnComplete) => {
-  const clipboardContents = await navigator.clipboard.read();
-  if (clipboardContents.length > 0) {
-    if (clipboardContents[0].types.includes("image/png")) {
-      clipboardContents[0].getType("image/png").then(blob => {
-        const blobToFile = new File([blob], "clipboard-image");
-        gImageMap.addFileImage(blobToFile).then(image => {
-          onComplete({ file: blobToFile, image });
-        });
-      });
+export const pasteClipboardImage = async (imageData: any, onComplete: OnComplete) => {
+  if (imageData.image) {
+    const blobToFile = new File([imageData.image], "clipboard-image.png");
+    gImageMap.addFileImage(blobToFile).then(image => {
+      onComplete({ image });
+    });
+  } else if (imageData.text) {
+    const url = imageData.text.match(/curriculum\/([^/]+\/images\/.*)/);
+    if (!url) {
+      console.error("ERROR: invalid image URL");
```

Comment:
It'd me more useful if the invalid url was part of the message, so we can figure out what is invalid about it.